### PR TITLE
Update AOT toolchain target to NetCoreAppToolCurrent

### LINKED ISSUE #68595: ILCompiler.DependencyAnalysisFramework targets net5.0 which is out-of-support soon
@trylek @MichalStrehovsky

Noticed that because it's the only project in the repo that depends on the System.Collections.Immutable package. If the project would upgrade to 6.0 we could also eliminate the package reference and the entry in Versions.props.

### PR DESCRIPTION
**Pull request by jkotas**
Fixes #68595

### CODE REVIEW

File: src/coreclr/tools/aot/ILCompiler.DependencyAnalysisFramework/ILCompiler.DependencyAnalysisFramework.csproj [hunk 1]
```diff
@@ -15,11 +15,6 @@
     <GenerateDependencyFile>false</GenerateDependencyFile>
     <Configurations>Debug;Release;Checked</Configurations>
   </PropertyGroup>
-  <ItemGroup>
-    <PackageReference Include="System.Collections.Immutable">
-      <Version>$(SystemCollectionsImmutableVersion)</Version>
```
Comment (am11) [inline src/coreclr/tools/aot/ILCompiler.DependencyAnalysisFramework/ILCompiler.DependencyAnalysisFramework.csproj:20]: This was the last usage of `SystemCollectionsImmutableVersion`, we can delete it from https://github.com/dotnet/runtime/blob/8f7761af3763d0b84acefbb117ed6becd83061d1/eng/Versions.props#L97

File: src/coreclr/tools/aot/ILCompiler/repro/repro.csproj [hunk 2]
```diff
@@ -1,6 +1,6 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFramework>net6.0</TargetFramework>
+    <TargetFramework>>$(NetCoreAppToolCurrent)</TargetFramework>
```
Comment (am11) [inline src/coreclr/tools/aot/ILCompiler/repro/repro.csproj:3]:
```suggestion
    <TargetFramework>$(NetCoreAppToolCurrent)</TargetFramework>
```

**Review by ViktorHofer — APPROVED** [review]
LGTM. Double checked by searching for other  unnecessary packagerefs and couldn't find any.

**Review by trylek — APPROVED** [review]
LGTM, thank you!

**Comment by jkotas** [comment]
All failures are unrelated infrastructure issues.

**Pull request merged**